crates/core/tedge_agent/src/tedge_to_te_converter/actor.rs

```diff
@@ -0,0 +1,236 @@
+        };
+
+        let msg = MqttMessage::new(&te_topic, message.payload_str()?).with_qos(message.qos);
+        self.messages.send(msg).await?;
+        Ok(())
+    }
+
+    // tedge/alarms/severity/alarm_type -> te/device/main///a/alarm_type, put severity in payload
+    // tedge/alarms/severity/child/alarm_type ->  te/device/child///a/alarm_type, put severity in payload
+    async fn convert_alarm(&mut self, message: MqttMessage) -> Result<(), TedgetoTeConverterError> {
+        let (cid, alarm_type, severity) =
+            get_child_id_and_alarm_type_and_severity(&message.topic.name)?;
+        let mut alarm: ThinEdgeAlarmData = serde_json::from_str(message.payload_str()?)?;
+        alarm.severity = severity.into();
+
+        let te_topic = match cid {
+            Some(cid) => Topic::new_unchecked(format!("te/device/{cid}///a/{alarm_type}").as_str()),
+            None => Topic::new_unchecked(format!("te/device/main///a/{alarm_type}").as_str()),
+        };
+
+        let msg = MqttMessage::new(&te_topic, serde_json::to_string(&alarm)?).with_qos(message.qos);
+        self.messages.send(msg).await?;
+        Ok(())
+    }
+
+    // tedge/events/event_type -> te/device/main///e/event_type
+    // tedge/events/child/event_type -> te/device/child///e/event_type
+    async fn convert_event(&mut self, message: MqttMessage) -> Result<(), TedgetoTeConverterError> {
+        let (cid, event_type) = get_child_id_and_event_type(&message.topic.name)?;
+        let te_topic = match cid {
+            Some(cid) => Topic::new_unchecked(format!("te/device/{cid}///e/{event_type}").as_str()),
+            None => Topic::new_unchecked(format!("te/device/main///e/{event_type}").as_str()),
+        };
+
+        let msg = MqttMessage::new(&te_topic, message.payload_str().unwrap()).with_qos(message.qos);
+        self.messages.send(msg).await?;
+        Ok(())
+    }
+
+    // tedge/health/service-name -> te/device/main/service/<service-name>/status/health
+    // tedge/health/child/service-name -> te/device/child/service/<service-name>/status/health
+    async fn convert_health_status_message(
+        &mut self,
+        message: MqttMessage,
+    ) -> Result<(), TedgetoTeConverterError> {
+        let (cid, service_name) = get_child_id_and_service_name(&message.topic.name)?;
+        let te_topic = match cid {
+            Some(cid) => Topic::new_unchecked(
+                format!("te/device/{cid}/service/{service_name}/status/health").as_str(),
+            ),
+            None => Topic::new_unchecked(
+                format!("te/device/main/service/{service_name}/status/health").as_str(),
+            ),
+        };
+        let msg = MqttMessage::new(&te_topic, message.payload_str()?).with_qos(message.qos);
+
+        self.messages.send(msg).await?;
+        Ok(())
+    }
+}
+
+fn get_child_id_and_event_type(
+    topic: &str,
+) -> Result<(Option<&str>, &str), TedgetoTeConverterError> {
```

Comment:
I don't think this function helps.

- It's used in a single place: `convert_event()`.
- The returned values are meaningful only in the context of `convert_event()`.
- The errors are artificial.